pallets/pool-system/src/impls.rs

```diff
@@ -48,6 +48,7 @@ impl<T: Config> PoolInspect<T::AccountId, T::CurrencyId> for Pallet<T> {
 		// Get cached nav as calculating current nav would be too computationally expensive
 		let (nav, nav_last_updated) = T::NAV::nav(pool_id)?;
 		let total_assets = pool.reserve.total.saturating_add(nav);
+		// why not checked/ensure?
```

Comment:
I'm afraid that the `saturating_ops` could lead to breaking some invariants. In this case, if, for some reason, the `nav` value is very big (i.e. an issue on the `NAV` side), `pool-system` will not be able to realize that and will successfully compute this when it shouldn't.

Reply:
Agreed, this should be a checked rather than saturating add

Reply:
Good catch. @lemunozm should I create clean up issues for this or do you wanna do this after we closed this review pr?

Reply:
Like you want 😃 . If in some time there are no created issues from this. I can go through this review PR and create the required issues, to ensure the important emerging points are caught.

Reply:
I Hope i will have time tomorrow for creating them.
Will leave a comment here if it is done.

Reply:
Done.

Reply:
https://github.com/centrifuge/centrifuge-chain/issues/1165